amqprs/src/net/reader_handler.rs

```diff
@@ -291,9 +291,11 @@ impl ReaderHandler {
 
                         // should call self.io_failure_notify.notify_one();?
                         #[cfg(feature="traces")]
-                        error!("missing heartbeat from server for {}", self.amqp_connection);
+                        warn!("missing heartbeat from server for {}", self.amqp_connection);
```

Comment:
can you also log the current missed heartbeats and the max allowed heartbeat, e.g. "{heartbeat_miss}/{MAX_HEARTBEAT_MISS}"

Reply:
`MAX_HEARTBEAT_MISS` must be set to `2` per AMQP 0-9-1 protocol spec. The second missed heartbeat must be treated as an indication of an unresponsive peer.

So how much value would there be with logging `1/2` and `2/2` like that?

Reply:
1/3, 2/3, 3/3 
I set to 3 as a best guess, but I think it better to be configured parameter, because it is also relevant the heatbeat interval settings

Reply:
@gftea the spec is very explicit: two missed heartbeats mean the peer must be considered unresponsive. This is very explicitly not a configurable parameter in any other client I can think of or RabbitMQ itself.

Reply:
In practice, if user config heatbeat interval to higher or lower value, do you still think 2 missed heartbeats hardcode limit still applies? make it configable with default. according to spec, and allow user to tune it based on their use cases will be flexible

Reply:
It does not matter what I think, the rest of the ecosystem uses two missed heartbeats for threshold and two only.

The docs [mention that very low values will produce false positives](https://www.rabbitmq.com/docs/heartbeats#false-positives).

Reply:
@gftea That sounds good. I can change the value to 2 as well, matching up with the spec. Not sure how atomic you want commits/pr's to be.

Reply:
@RoloEdits , thanks, yes, lets update to 2.
You can just push new commit, when merge, would be squashed